File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    
    int maxX = 0;
    int maxY = 0;
    for(int i = 0; i < sizes.size(); i++) {
        vector<int> cur = sizes[i];
        
        if(cur[0] < cur[1]) {
            int temp = cur[0];
            cur[0] = cur [1];
            cur[1] = temp;
        }
        
        if(cur[0] > maxX) maxX = cur[0];
        if(cur[1] > maxY) maxY = cur[1];
    }
    
    answer = maxX * maxY;
    return answer;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cs
using System;

public class Solution {
    public int solution(int[,] sizes) {        
        int xMax = 0;
        int yMax = 0;
        
        for(int i = 0; i < sizes.GetLength(0); i++) {
            if(sizes[i, 0] < sizes[i,1]){
                int temp = sizes[i,1];
                sizes[i,1] = sizes[i,0];
                sizes[i,0] = temp;                
            }
            
            if(sizes[i,0] > xMax) xMax = sizes[i,0];
            if(sizes[i,1] > yMax) yMax = sizes[i,1];
        }
        
        return xMax * yMax;
    }
}